trunk/modules/python/dataset/reader.py

```diff
@@ -22,6 +22,13 @@ def mostCommonExt(files, samplesize=None) :
 
 def padImageData(imgData, dims) :
     '''Add zeropadding to an image to achieve the target dimensions.'''
+    if type(dims) != tuple :
+        dims = tuple(dims)
+
+    # add an extra singleton dimension for 1-channel arrays
+    if len(imgData.shape) == (len(dims) - 1) :
+        imgData = np.expand_dims(imgData, axis=0)
+
     if (imgData.shape != dims) :
```

Comment:
Here just do a "if image.shape != tuple(dims) :".  Just casting to tuple will clean up the code a bit, and no parenthesis.